src/client-side-metrics/exporter.ts

```diff
@@ -0,0 +1,409 @@
+ *   object contains resource attributes, scope information, and a list of
+ *   metrics with their associated data points.
+ *
+ * @returns An object representing a `CreateTimeSeriesRequest`, ready for sending
+ *   to the Google Cloud Monitoring API. This object contains the project name
+ *   and an array of time series data points, formatted for ingestion by
+ *   Cloud Monitoring.
+ *
+ * @throws Will throw an error if there are issues converting the data.
+ *
+ * @remarks
+ *   The output format is specific to the Cloud Monitoring API and involves
+ *   mapping OpenTelemetry concepts to Cloud Monitoring's data model, including:
+ *   - Mapping resource attributes to resource labels.
+ *   - Mapping metric attributes to metric labels.
+ *   - Handling different metric types (counter, distribution).
+ *   - Converting data points to the correct structure, including start and end
+ *     times, values, and bucket information for distributions.
+ *
+ * @example
+ *   const exportInput: ExportInput = { ... }; // Example ExportInput object
+ *   const monitoringRequest = metricsToRequest(exportInput);
+ *   // monitoringRequest can now be used in monitoringClient.createTimeSeries(monitoringRequest)
+ *
+ *
+ */
+export function metricsToRequest(exportArgs: ExportInput) {
+  const timeSeriesArray = [];
+  const resourceLabels = {
+    cluster: exportArgs.resource._syncAttributes['monitored_resource.cluster'],
+    instance:
+      exportArgs.resource._syncAttributes['monitored_resource.instance_id'],
+    project_id:
+      exportArgs.resource._syncAttributes['monitored_resource.project_id'],
+    table: exportArgs.resource._syncAttributes['monitored_resource.table'],
+    zone: exportArgs.resource._syncAttributes['monitored_resource.zone'],
+  };
+  for (const scopeMetrics of exportArgs.scopeMetrics) {
+    for (const scopeMetric of scopeMetrics.metrics) {
+      const metricName = scopeMetric.descriptor.name;
+      for (const dataPoint of scopeMetric.dataPoints) {
+        const value = dataPoint.value;
+        const allAttributes = dataPoint.attributes;
+        const streaming = allAttributes.streamingOperation;
+        /*
+        metricLabels are built from the open telemetry attributes that are set
+        when a data point is recorded. This means that for one metric there may
+        be multiple time series' with different attributes, but the resource
+        labels will always be the same for a particular export call.
+         */
+        const metricLabels = Object.assign(
+          {
+            app_profile: allAttributes.appProfileId,
+            client_name: allAttributes.clientName,
+            method: allAttributes.methodName,
+            status:
+              (allAttributes as OnAttemptAttribute).attemptStatus?.toString() ??
+              (
+                allAttributes as OnOperationAttribute
+              ).finalOperationStatus?.toString(),
```

Comment:
instead of doing toString here, can't you write it as a string to start with?

Reply:
Yes. I pushed the string conversion back to the metrics collector [here](https://github.com/googleapis/nodejs-bigtable/pull/1569/commits/a587d0c97b83b061b20016cf5edb062049284d10) and use status now instead of attemptStatus and finalOperationStatus.